test/ymlr/encode_test.exs

```diff
@@ -18,6 +18,14 @@ defmodule Ymlr.EncodeTest do
     test "plain strings" do
       assert_identity_and_output("hello world", "hello world")
       assert_identity_and_output("that's it", "that's it")
+
+      # Strings where the ":", "?", or "-" indicator character is used as the
+      # first character may be encoded in Plain Style (without being quoted)
+      # if followed by a non-space “safe” character.
+      # https://yaml.org/spec/1.2.2/#733-plain-style
+      assert_identity_and_output("?x", ~S(?x))
+      assert_identity_and_output(":x", ~S(:x))
+      assert_identity_and_output("-x", ~S(-x))
```

Comment:
If we're expanding test coverage to handle some of these corner cases pertaining to indicator characters, then I'd also propose adding these tests.